hack/build-manifests.sh

```diff
@@ -27,19 +27,51 @@ manifest_docker_prefix=${manifest_docker_prefix-${docker_prefix}}
 args=$(cd ${KUBEVIRT_DIR}/manifests && find * -type f -name "*.yaml.in")
 
 rm -rf ${MANIFESTS_OUT_DIR}
+rm -rf ${MANIFEST_TEMPLATES_OUT_DIR}
 
 (cd ${KUBEVIRT_DIR}/tools/manifest-templator/ && go build)
 
 for arg in $args; do
     final_out_dir=$(dirname ${MANIFESTS_OUT_DIR}/${arg})
+    final_templates_out_dir=$(dirname ${MANIFEST_TEMPLATES_OUT_DIR}/${arg})
     mkdir -p ${final_out_dir}
+    mkdir -p ${final_templates_out_dir}
     manifest=$(basename -s .in ${arg})
     outfile=${final_out_dir}/${manifest}
+    template_outfile=${final_templates_out_dir}/${manifest}.j2
 
     ${KUBEVIRT_DIR}/tools/manifest-templator/manifest-templator \
         --namespace=${namespace} \
         --docker-prefix=${manifest_docker_prefix} \
         --docker-tag=${docker_tag} \
         --generated-manifests-dir=${KUBEVIRT_DIR}/manifests/generated/ \
         --input-file=${KUBEVIRT_DIR}/manifests/$arg >${outfile}
+
+    ${KUBEVIRT_DIR}/tools/manifest-templator/manifest-templator \
+        --namespace="{{ namespace }}" \
+        --docker-prefix="{{ docker_prefix }}" \
+        --docker-tag="{{ docker_tag }}" \
+        --generated-manifests-dir=${KUBEVIRT_DIR}/manifests/generated/ \
+        --input-file=${KUBEVIRT_DIR}/manifests/$arg >${template_outfile}
+done
+
+# make sure that template manifests align with release manifests
+export namespace=${namespace}
+export docker_tag=${docker_tag}
+export docker_prefix=${manifest_docker_prefix}
+
+TMP_DIR=$(mktemp -d)
+cleanup() {
+    ret=$?
+    rm -rf "${TMP_DIR}"
+    exit ${ret}
+}
+trap "cleanup" INT TERM EXIT
+
+for file in $(find ${MANIFEST_TEMPLATES_OUT_DIR}/ -type f); do
+    mkdir -p ${TMP_DIR}/$(dirname ${file})
+    j2 ${file} | sed -e '/.$/a\' >${TMP_DIR}/${file%.j2}
```

Comment:
I think you are still going to have to pass j2 the variables? Otherwise the vars are rendered as blank and will fail a diff:
-e "namespace='${namespace}'" -e "docker_tag='${docker_tag}'" -e "docker_prefix='${manifest_docker_prefix}'"

Reply:
It is set. See the three exports above.

Reply:
Those are the environment variables, but I don't see how they get passed to j2. 

When I run hack/buld-manifests.sh, I get a bunch of output from the diff since they are not the same, e.g.:
<...snip..>
diff -r /tmp/kubevirt/_out/manifests/testing/rbac-for-testing.yaml /tmp/tmp.r3FGBWzDyR//tmp/kubevirt/_out/templates/manifests/testing/rbac-for-testing.yaml
6c6
<   namespace: kube-system
---
>   namespace: 
14c14
<   namespace: kube-system
---
>   namespace: 
24c24
<     namespace: kube-system
---
>     namespace: 
<..end...>

Maybe it runs different in a container?

Reply:
@dcritch are you using the latest version of my branch? It works for me inside and outside of the container and it also seems to work on travis.

Reply:
I am.

Weird.... I was trying on Fedora 27. On a whim I tried RHEL7 and it is failing there too.

Reply:
Maybe its my j2 binary? I 'm using 1.2.1 from pip

Reply:
Seems so. I have 0.3.1.

Reply:
Oh I think you have another client tool entirely. Mine is called `j2cli`, but the resulting binary after installation from pip is called `j2`.

Reply:
That was it. I installed j2cli instead and the script runs clean now. Sorry for the noise!